Relax tight dependencies that upstream uses to keep MSRV low

Upstream is concerned with keeping the MSRV low, but we don't need to
commit to that same approach to resolving the concern (debian archive
management should handle it for us).

See also upstream comments in
https://gitlab.com/sequoia-pgp/sequoia/-/blob/master/openpgp/Cargo.toml

Signed-off-by: Daniel Kahn Gillmor <dkg@fifthhorseman.net>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,7 +32,7 @@
 version = "1"
 
 [dependencies.backtrace]
-version = "= 0.3.46"
+version = "0.3"
 
 [dependencies.base64]
 version = ">= 0.11, < 0.13"
@@ -46,7 +46,7 @@
 optional = true
 
 [dependencies.dyn-clone]
-version = ">= 1.0.1, < 1.0.2"
+version = "1"
 
 [dependencies.flate2]
 version = "1.0.1"
@@ -85,17 +85,17 @@
 version = "1"
 
 [dependencies.unicode-normalization]
-version = "= 0.1.9"
+version = "0.1.9"
 [dev-dependencies.quickcheck]
 version = "0.9"
 default-features = false
 
 [dev-dependencies.rand]
 version = "0.7"
 default-features = false
 
 [dev-dependencies.rpassword]
-version = "=4.0.3"
+version = "4.0.3"
 [build-dependencies.lalrpop]
 version = "0.17"
 
